Sources/Network/Network.swift

```diff
@@ -5,18 +5,35 @@ public enum Network {
 
     // MARK: - TypeAlias
 
-    public typealias CompletionClosure<R> = (Result<R, Error>) -> Void
+    public typealias CompletionClosure<R> = (Result<Value<R>, Error>) -> Void
     public typealias AuthenticationCompletionClosure = (URLSession.AuthChallengeDisposition, URLCredential?) -> Void
 
-    // MARK: - Network Error
+    // MARK: - Response value
 
-    public enum Error: Swift.Error {
-        case http(code: HTTP.StatusCode, apiError: Swift.Error?)
-        case noData
-        case url(Swift.Error)
-        case badResponse
-        case authenticator(Swift.Error)
-        case retry(errors: [Swift.Error], totalDelay: ResourceRetry.Delay, retryError: ResourceRetry.Error)
+    public struct Value<R> {
+
+        let value: R
+
+        let response: URLResponse
+    }
+
+    // MARK: - Response error
+
+    public struct Error: Swift.Error {
```

Comment:
If we have to make this conform to `Swift.Error` (because of `Result`), why not simply add a `response: URLResponse?` to all the enum cases of the `Error` (now `ErrorType`)? 

We could then make a simple computed property to extract it.

Having these two error "levels" feels a bit weird...

Reply:
Yeah, it does feel weird 😕 
I'll try the other approach